state-chain/cf-integration-tests/src/network.rs

```diff
@@ -215,10 +218,12 @@ impl Engine {
 
 	// Handle events coming in from the state chain
 	// TODO have this abstracted out
-	fn handle_state_chain_events(&mut self, events: &[RuntimeEvent]) {
+	fn handle_state_chain_events(&mut self, events: &[RuntimeEvent], cfe_events: &[CfeEvent]) {
 		if self.live {
 			// Being a CurrentAuthority we would respond to certain events
 			if self.state() == ChainflipAccountState::CurrentAuthority {
+				// Note: these aren't events that the engine normally responds to, but
+				// we process them here due to the way integration tests are set up
 				on_events! {
 					events,
 					RuntimeEvent::Validator(
```

Comment:
This is actually outdated now that we optimistically rotate, the CFE should use the new key after the key activation begins.

Reply:
I had a go at this, and it doesn't seem trivial to me. I tried using `AwaitingGovernanceActivation` instead, but then it is not emitted for Ethereum, so it would probably need a separate mechanism. I'm not even convinced that this would have been more accurate, since CFEs doesn't have a notion of "active" key. They just save every key they generate immediately and then it is up to the SC to decide when, if ever, to use them (so maybe having a map from pubkey to secret key would be more accurate). I think it is OK to leave it as is (and maybe make some improvements to integration tests in a separate PR).